Formula/png++.rb

```diff
@@ -11,9 +12,55 @@ class Pngxx < Formula
     sha256 "bed85536ebfec14b98b056e874d116527f154eff2b77024aee52fc407d66791f" => :sierra
   end
 
+  depends_on "imagemagick" => :build
   depends_on "libpng"
 
+  # Issues with GNU strerror_r being used because Darwin libc does not define _POSIX_C_SOURCE
+  # Also patches the test script to use imagemagick instead of binary comparison
+  patch :DATA
```

Comment:
Where is the `strerror_r` patch coming from?

Reply:
Internal testing. The original code is taken from the GNU strerror_r page as a way to test the existence of `strerror_r`, but it fails on MacOS if `_POSIX_C_SOURCE` is not defined, so it uses the GNU version of the function instead. The actual logic is flawed because the `_POSIX_C_SOURCE` needs to be defined by an application to select a certain version of POSIX conformity, not tested against to see which version is provided by the libc. gcc will define `_GNU_SOURCE`, which will then define `_POSIX_C_SOURCE` in the glibc, forcing the highest POSIX level by default. Selecting the GNU or non-GNU function is then not related to the POSIX level.

Even if the `_POSIX_C_SOURCE` trick was correct, png++ does not provide another mechanism for lower POSIX versions. For this reason, I did not want to implement the full logic and just went with a switch for GNU or non-GNU (XSI, QNX, BSD, Darwin and Android Bionic from a quick survey).

I do blame MacOS for not providing the `_POSIX_C_SOURCE` macro by default like gcc, though, because it defines `__DARWIN_C_LEVEL` instead. So the actual fix would have been to test `_POSIX_C_SOURCE` and `__DARWIN_C_LEVEL`, then select GNU or non-GNU, otherwise define a new mechanism.

Reply:
Was the issue reported upstream? Have they accepted this fix?

Reply:
A similar issue was reported with a different fix, but with the same result. Non-GNU got to use the non-GNU version, GNU used GNU version and POSIX checks were short-circuited instead of removed. That patch was sent four years ago to the source owner on Savannah.nongnu.org and has been in limbo ever since. The owner replied two years ago saying that development was still active on Savannah, even though no release has been made in four years. I can try to submit this patch instead, if you want.

Reply:
After some thought, I will implement the Savannah patch directly and poke the owner to make sure they are still alive.